nestkernel/connection_manager.cpp

```diff
@@ -1432,9 +1437,13 @@ nest::ConnectionManager::set_have_connections_changed( const thread tid )
   // performance issues on supercomputers.
   if ( have_connections_changed_[ tid ].is_false() )
   {
-    std::string msg =
-      "New connections created, connection descriptors previously obtained using 'GetConnections' are now invalid.";
-    LOG( M_WARNING, "ConnectionManager", msg );
+    if ( has_get_connections_been_called_ )
+    {
+      std::string msg =
+        "New connections created, connection descriptors previously obtained using 'GetConnections' are now invalid.";
+      LOG( M_WARNING, "ConnectionManager", msg );
+      set_has_get_connections_been_called( false );
```

Comment:
Am I right in that this line is here to make sure the warning is only printed once? I wonder if we should have a second flag `invalid_connection_descriptor_warning_printed_` for this, as the flag is technically in a wrong state after this statement. With the current set of changes, I'm pretty sure that nothing bad can happen, but if people start relying on `has_get_connections_been_called_` for other purposes in the future, this might blow up and lead to errors that could potentially be hard to debug. But maybe I'm being overly paranoid here...

@stinebuu: any opinions? How are you dealing with this in the deprecation warning framework?

Reply:
@jougs For the deprecation warnings I have a map between the name of deprecated functions and whether they are deprecated. I also set them to `false` after the warning is issued, but it is clear that the bool is connected to a deprecation warning. 
https://github.com/nest/nest-simulator/blob/47b7e43f72317911b3e285daa9ecd4f45688fa5d/nestkernel/deprecation_warning.cpp#L37-L45

I think you are right and that `has_get_connections_been_called_` might be used for other purposes in the future, but it might be enough with just one bool if we rename it? Because it is only used for this now, so I do not want to have a bool laying around that is not used at all.

@hakonsbm I think you should also add a comment after `set_has_get_connections_been_called( false )` just to point out why it is there ;)

Reply:
I also think that if the flag starts being used for other things it should be renamed. Until then, I've added some comments to clarify the usage of `set_has_get_connections_been_called()`.

Reply:
Fine by me!